harbor-registry.yaml

```diff
@@ -0,0 +1,56 @@
+package:
+  name: harbor-registry
+  version: 2.8.3
+  epoch: 0
+  description: An open source trusted cloud native registry project that stores, signs, and scans content (registry)
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  environment:
+    DISTRIBUTION_DIR: "/usr/lib/go/src/github.com/docker/distribution"
+    CGO_ENABLED: "0"
+    GO111MODULE: "auto"
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/distribution/distribution
+      tag: v${{package.version}}
+      expected-commit: 4772604ae973031ab32dd9805a4bccf61d94909f
+
+  # https://github.com/goharbor/harbor/blob/aa1f79cd0c5115667a8f3f78ea963ec2623f501e/make/photon/registry/redis.patch
+  - uses: patch
+    with:
+      patches: redis.patch
+
+  - runs: |
+      # Adds source modules to $GOPATH
+      mkdir -p "${DISTRIBUTION_DIR}"
+      cp -rf . "${DISTRIBUTION_DIR}"
+
+  - uses: go/build
+    with:
+      packages: ./cmd/registry
+      output: harbor-registry
+
+  - runs: |
+      mkdir -p ${{targets.destdir}}/usr/bin
+      mkdir -p ${{targets.destdir}}/etc/registry
+
+      # Symlink to expected binary path
+      ln -sf /usr/bin/harbor-registry ${{targets.destdir}}/usr/bin/registry_DO_NOT_USE_GC
+      # Use example config as registry config
+      cp ./cmd/registry/config-example.yml ${{targets.destdir}}/etc/registry/config.yml
+
+test:
+  pipeline:
+    - runs: |
+        # The registry should start and begin listening before it's killed
+        registry_DO_NOT_USE_GC serve /etc/registry/config.yml & sleep 5; kill $!
+
```

Comment:
you can enhance the test by hitting the endpoint and expecting a 200 OK response back on 
localhost:5000

Reply:
Thanks for the suggestion. Will get that up

Reply:
Added